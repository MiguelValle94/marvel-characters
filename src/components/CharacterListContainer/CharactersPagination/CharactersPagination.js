import React from 'react';
import PropTypes from 'prop-types';

import Pagination from './Pagination';

import './CharactersPagination.scss';

export default function CharactersPagination({ page, pages, onSelectPage }) {
  return (
    <div className='pagination__wrapper'>
      <Pagination>
        <Pagination.Prev
          disabled={page === 1}
          onClick={() => onSelectPage(page - 1)}
        />
        {page !== 1 && (
          <Pagination.Item onClick={() => onSelectPage(1)}>1</Pagination.Item>
        )}
        {page > 3 && <Pagination.Ellipsis disabled />}
        {page > 2 && (
          <Pagination.Item onClick={() => onSelectPage(page - 1)}>
            {page - 1}
          </Pagination.Item>
        )}
        <Pagination.Item active>{page}</Pagination.Item>
        {page > 2 && page < pages && page + 1 !== pages && (
          <Pagination.Item onClick={() => onSelectPage(page + 1)}>
            {page + 1}
          </Pagination.Item>
        )}
        {page < pages - 1 && page + 1 !== pages - 1 && (
          <Pagination.Ellipsis disabled />
        )}
        {page < pages && (
          <Pagination.Item onClick={() => onSelectPage(pages)}>
            {pages}
          </Pagination.Item>
        )}
        <Pagination.Next
          disabled={page === pages}
          onClick={() => onSelectPage(page + 1)}
        />
      </Pagination>
    </div>
  );
}

CharactersPagination.propTypes = {
  page: PropTypes.number.isRequired,
  pages: PropTypes.number.isRequired,
  onSelectPage: PropTypes.func.isRequired
};
