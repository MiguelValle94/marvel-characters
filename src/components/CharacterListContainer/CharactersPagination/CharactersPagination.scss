.pagination {
  margin: 0;
  padding: 10px;

  &__wrapper {
    display: flex;
    justify-content: flex-end;
  }
}
