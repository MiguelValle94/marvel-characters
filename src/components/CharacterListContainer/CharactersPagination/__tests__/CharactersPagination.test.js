import React from 'react';
import { shallow } from 'enzyme';

import CharactersPagination from '../CharactersPagination';

const setUpShallow = (page, pages, handleSelectPage) => {
  const component = shallow(
    <CharactersPagination
      page={page}
      pages={pages}
      onSelectPage={handleSelectPage}
    />
  );
  return component;
};

describe('CharactersPagination', () => {
  let component;
  const handleSelectPage = jest.fn();

  it('should display pagination with given properties', () => {
    component = setUpShallow(1, 2, handleSelectPage);
    expect(component).toMatchSnapshot();
  });
});
