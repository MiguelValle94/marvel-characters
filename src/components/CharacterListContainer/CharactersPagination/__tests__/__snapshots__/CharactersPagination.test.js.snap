// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersPagination should display pagination with given properties 1`] = `
<div
  className="pagination__wrapper"
>
  <ForwardRef>
    <Prev
      disabled={true}
      onClick={[Function]}
    />
    <PageItem
      active={true}
      activeLabel="(current)"
      disabled={false}
    >
      1
    </PageItem>
    <PageItem
      active={false}
      activeLabel="(current)"
      disabled={false}
      onClick={[Function]}
    >
      2
    </PageItem>
    <Next
      disabled={false}
      onClick={[Function]}
    />
  </ForwardRef>
</div>
`;
